test/extensions/transport_sockets/proxy_protocol/proxy_protocol_integration_test.cc

```diff
@@ -398,5 +400,239 @@ TEST_P(ProxyProtocolHttpIntegrationTest, TestProxyProtocolHealthCheck) {
+      envoy::extensions::filters::listener::proxy_protocol::v3::ProxyProtocol proxy_protocol;
+      auto pass_through_tlvs = proxy_protocol.mutable_pass_through_tlvs();
+      if (pass_all_tlvs_) {
+        pass_through_tlvs->set_match_type(ProxyProtocolPassThroughTLVs::INCLUDE_ALL);
+      } else {
+        pass_through_tlvs->set_match_type(ProxyProtocolPassThroughTLVs::INCLUDE);
+        for (const auto& tlv_type : tlvs_listener_) {
+          pass_through_tlvs->add_tlv_type(tlv_type);
+        }
+      }
+
+      auto* listener = bootstrap.mutable_static_resources()->mutable_listeners(0);
+      auto* ppv_filter = listener->add_listener_filters();
+      ppv_filter->set_name("envoy.listener.proxy_protocol");
+      ppv_filter->mutable_typed_config()->PackFrom(proxy_protocol);
+    });
+
+    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      auto* transport_socket =
+          bootstrap.mutable_static_resources()->mutable_clusters(0)->mutable_transport_socket();
+      transport_socket->set_name("envoy.transport_sockets.upstream_proxy_protocol");
+      envoy::config::core::v3::TransportSocket inner_socket;
+      inner_socket.set_name("envoy.transport_sockets.raw_buffer");
+
+      envoy::config::core::v3::ProxyProtocolConfig proxy_protocol;
+      proxy_protocol.set_version(envoy::config::core::v3::ProxyProtocolConfig::V2);
+      auto pass_through_tlvs = proxy_protocol.mutable_pass_through_tlvs();
+
+      if (pass_all_tlvs_) {
+        pass_through_tlvs->set_match_type(ProxyProtocolPassThroughTLVs::INCLUDE_ALL);
+      } else {
+        pass_through_tlvs->set_match_type(ProxyProtocolPassThroughTLVs::INCLUDE);
+        for (const auto& tlv_type : tlvs_upstream_) {
+          pass_through_tlvs->add_tlv_type(tlv_type);
+        }
+      }
+
+      envoy::extensions::transport_sockets::proxy_protocol::v3::ProxyProtocolUpstreamTransport
+          proxy_proto_transport;
+      proxy_proto_transport.mutable_transport_socket()->MergeFrom(inner_socket);
+      proxy_proto_transport.mutable_config()->MergeFrom(proxy_protocol);
+      transport_socket->mutable_typed_config()->PackFrom(proxy_proto_transport);
+    });
+
+    BaseIntegrationTest::initialize();
+  }
+
+  FakeRawConnectionPtr fake_upstream_connection_;
+
+private:
+  bool pass_all_tlvs_ = false;
+  std::vector<uint8_t> tlvs_listener_;
+  std::vector<uint8_t> tlvs_upstream_;
+};
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, ProxyProtocolTLVsIntegrationTest,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+
+// This test adding the listener proxy protocol filter and upstream proxy filter, the TLVs
+// are passed by listener and re-generated in transport socket based on API config.
+TEST_P(ProxyProtocolTLVsIntegrationTest, TestV2TLVProxyProtocol_PassSepcificTLVs) {
```

Comment:
You do not need to add `_` in tests names per style.

Reply:
Done.